packages/forms/backend/sirius-components-collaborative-forms/src/main/java/org/eclipse/sirius/components/collaborative/forms/handlers/EditTextfieldEventHandler.java

```diff
@@ -101,7 +101,11 @@ public void handle(One<IPayload> payloadSink, Many<ChangeDescription> changeDesc
             }
             if (status instanceof Success success) {
                 payload = new SuccessPayload(formInput.id(), success.getMessages());
-                changeDescription = new ChangeDescription(ChangeKind.SEMANTIC_CHANGE, formInput.representationId(), formInput);
+                if (success.getChangeKind().isBlank()) {
+                    changeDescription = new ChangeDescription(ChangeKind.SEMANTIC_CHANGE, formInput.representationId(), formInput);
+                } else {
+                    changeDescription = new ChangeDescription(success.getChangeKind(), formInput.representationId(), formInput, success.getParameters());
+                }
```

Comment:
If I understand things correctly, only the second behavior should exists:

```
new ChangeDescription(success.getChangeKind(), formInput.representationId(), formInput, success.getParameters());
```

but for compatibility reasons, you have also kept the previous one

```
new ChangeDescription(ChangeKind.SEMANTIC_CHANGE, formInput.representationId(), formInput);
```

That's fine for now BUT we will indicate in the CHANGELOG that we will move exclusively to the new behavior and won't emit a SEMANTIC_CHANGE by default for all widgets of the form DSL. Specifiers will have to decide what kind of change they want to emit.